distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/config/ConfigureTestCommons.java

```diff
@@ -13,71 +13,110 @@
  */
 package org.codice.ddf.itests.common.config;
 
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.with;
+
 import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.Dictionary;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
 import org.codice.ddf.configuration.DictionaryMap;
 import org.codice.ddf.itests.common.AdminConfig;
 import org.osgi.service.cm.Configuration;
 
 public class ConfigureTestCommons {
 
-  public static final String METACARD_VALIDITITY_FILTER_PLUGIN_SERVICE_PID =
+  private static final String METACARD_VALIDITY_FILTER_PLUGIN_SERVICE_PID =
       "ddf.catalog.metacard.validation.MetacardValidityFilterPlugin";
 
-  public static final String METACARD_VALIDITITY_MARKER_PLUGIN_SERVICE_PID =
+  private static final String METACARD_VALIDITY_MARKER_PLUGIN_SERVICE_PID =
       "ddf.catalog.metacard.validation.MetacardValidityMarkerPlugin";
 
-  public static final String METACARD_ATTRIBUTE_SECURITY_POLICY_PLUGIN_PID =
+  private static final String METACARD_ATTRIBUTE_SECURITY_POLICY_PLUGIN_PID =
       "org.codice.ddf.catalog.security.policy.metacard.MetacardAttributeSecurityPolicyPlugin";
 
-  public static final String AUTH_Z_REALM_PID = "ddf.security.pdp.realm.AuthzRealm";
+  private static final String AUTH_Z_REALM_PID = "ddf.security.pdp.realm.AuthzRealm";
 
   private ConfigureTestCommons() {}
 
-  public static void configureMetacardValidityFilterPlugin(
-      List<String> securityAttributeMappings, AdminConfig configAdmin) throws IOException {
-    Configuration config =
-        configAdmin.getConfiguration(METACARD_VALIDITITY_FILTER_PLUGIN_SERVICE_PID, null);
-    Dictionary<String, Object> properties = new DictionaryMap<>();
-    properties.put("attributeMap", securityAttributeMappings);
-    config.update(properties);
+  public static Dictionary<String, Object> configureService(
+      String pid, Dictionary<String, Object> props, AdminConfig adminConfig) throws IOException {
+    Configuration config = adminConfig.getConfiguration(pid, null);
+    Dictionary<String, Object> oldProps = config.getProperties();
+    config.update(props);
+
+    // verify update
+    with()
+        .pollInterval(1, SECONDS)
+        .await()
+        .atMost(TimeUnit.MINUTES.toSeconds(2), SECONDS)
+        .ignoreExceptions()
+        .until(
+            () -> {
+              Dictionary<String, Object> savedProps = config.getProperties();
+              List<String> newKeys = Collections.list(props.keys());
+              for (String key : newKeys) {
+                Object savedValue = savedProps.get(key);
+                Object incomingValue = props.get(key);
+                if (incomingValue.getClass().isArray() && savedValue.getClass().isArray()) {
+                  if (!Arrays.equals((Object[]) savedValue, (Object[]) incomingValue)) {
```

Comment:
❓ Is it guarantee that it would be an array of object?

Here is a suggested implementation of an `equals()` for 2 dictionaries:

```
private static boolean equals(Dictionary<String, Object> x, Dictionary<String, Object> y) {
    if (x.size() != y.size()) {
      return false;
    }
    for (final Enumeration<String> e = x.keys(); e.hasMoreElements(); ) {
      final String key = e.nextElement();

      if (!Objects.deepEquals(x.get(key), y.get(key))) {
        return false;
      }
    }
    return true;
  }
```